astropy/io/fits/column.py

```diff
@@ -2271,6 +2271,16 @@ def _makep(array, descr_output, format, nrows=None):
         nelem = data_output[idx].shape
         descr_output[idx, 0] = np.prod(nelem)
         descr_output[idx, 1] = _offset
+
+        # detect overflow when using P format
+        if (
+            descr_output.dtype == np.int32
+            and int(descr_output[idx, 0]) * _nbytes > np.iinfo(descr_output.dtype).max
```

Comment:
I think I'd just put `>= 2**31` for clarity. What you have makes more sense if the dtype could vary here.

Reply:
Good point, thanks